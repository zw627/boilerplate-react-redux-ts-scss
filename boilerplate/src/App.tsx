import React from "react";

const App: React.FC = () => {
  return (
    <div>
      <h1>React-Redux-TypeScript-Sass boilerplate</h1>
    </div>
  );
};

export default App;
