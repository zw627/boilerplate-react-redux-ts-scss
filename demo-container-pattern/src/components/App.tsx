import React, { Suspense } from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import { Provider } from "react-redux";

import store from "Store/index";
import Home from "Components/Home";
import NotificationStorage from "Components/NotificationStorage";
import ThemeSwtich from "Containers/ThemeSwitch";

const Scoreboard = React.lazy(() => import("Components/Scoreboard"));
const NotFound = React.lazy(() => import("Components/NotFound"));

const App: React.FC<Record<string, unknown>> = () => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <Suspense fallback={<div>Loading...</div>}>
          <Switch>
            <Route exact path="/" component={Home} />
            <Route exact path="/scoreboard" component={Scoreboard} />
            <Route component={NotFound} />
          </Switch>
        </Suspense>
      </BrowserRouter>

      <NotificationStorage />
      <ThemeSwtich />
    </Provider>
  );
};

export default App;
