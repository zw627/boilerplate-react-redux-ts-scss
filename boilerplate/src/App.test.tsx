import React from "react";
import { mount, ReactWrapper } from "enzyme";

import App from "./App";

function setupWrapper(): ReactWrapper {
  return mount(<App />) as ReactWrapper;
}

describe("App", () => {
  let wrapper: ReactWrapper;

  beforeEach(() => {
    wrapper = setupWrapper();
  });

  test("compare to the last snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
