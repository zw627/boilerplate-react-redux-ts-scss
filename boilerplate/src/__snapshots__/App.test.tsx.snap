// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App compare to the last snapshot 1`] = `
<App>
  <div>
    <h1>
      React-Redux-TypeScript-Sass boilerplate
    </h1>
  </div>
</App>
`;
